docfx_project/tutorials/minimal-usage.md

```diff
@@ -0,0 +1,135 @@
+
+The configuration type is here `BaseConfig`, but you can use any class that extends `VersionedConfig`. The full range of available options in `BaseConfig` is shown in [config.example.yml](https://github.com/cognitedata/dotnet-extractor-utils/blob/master/ExtractorUtils/config/config.example.yml).
+
+A reasonable minimal configuration would look something like this:
+
+```yaml
+version: 1
+
+logger:
+  console:
+    level: "debug"
+
+cognite:
+    project: ${COGNITE_PROJECT}
+    host: ${COGNITE_BASE_URL}
+  
+    # This is for microsoft as IdP, to use a different provider,
+    # set implementation: Basic, and use token-url instead of tenant.
+    # See the example config for the full list of options.
+    idp-authentication:
+        # Directory tenant
+        tenant: ${COGNITE_TENANT_ID}
+        # Application Id
+        client-id: ${COGNITE_CLIENT_ID}
+        # Client secret
+        secret: ${COGNITE_CLIENT_SECRET}
+        # List of resource scopes, ex:
+        # scopes:
+        #   - scopeA
+        #   - scopeB
+        scopes:
+          - ${COGNITE_BASE_URL}/.default
+```
+
+Alternatively you can create the configuration object manually. For the equivalent to the minimal configuration above:
+
+```c#
+var config = new BaseConfig {
+    Version = 1,
+    Logger = new LoggerConfig {
+        Console = new ConsoleConfig {
+            Level = "debug"
+        }
+    },
+    Cognite = new CogniteConfig {
+        Project = Environment.GetEnvironmentVariable("COGNITE_PROJECT"),
+        Host = Environment.GetEnvironmentVariable("COGNITE_BASE_URL"),
+        IdpAuthentication = new AuthenticatorConfig
+        {
+            Tenant = Environment.GetEnvironmentVariable("COGNITE_TENANT"),
+            ClientId = Environment.GetEnvironmentVariable("COGNITE_CLIENT_ID"),
+            Secret = Environment.GetEnvironmentVariable("COGNITE_CLIENT_SECRET"),
+            Scopes = new[]
+            {
+                $"{Environment.GetEnvironmentVariable("COGNITE_BASE_URL")}/.default"
+            }
+        }
+    }
+};
+
+config.GenerateDefaults();
+
+services.AddExtractorDependencies(
+    configPath: null,
+    acceptedConfigVersions: new [] { 1 }, // Fail if "version" in the config file does not match this, this can be null to not have any version requirement.
+    appId: "my-app", // Application id passed with HTTP requests to CDF
+    userAgent: "myapp/1.0.0", // User agent for HTTP requests, including application version
+    addStateStore: false, // True if you want to register code for using a litedb state store
+    addLogger: true, // False if you do not want to register a serilog logger
+    addMetrics: false, // True if you want to push or serve metrics to prometheus
+    config: config // Pass a pre-existing config object
+);
+```
+
+## Using just Cognite.Extensions
```

Comment:
Could we have similar sections on using just upload queues?

Reply:
They are part of the full package, so you can't use them without the full list of dependencies, but a short guide on how to use them given a `destination` object would be nice.

Reply:
Looking at it, there already is one for each queue.